docs/src/main/sphinx/develop/connectors.rst

```diff
@@ -238,17 +238,148 @@ the strategy employed by the Example HTTP connector.
 .. _connector-record-set-provider:
 
 ConnectorRecordSetProvider
-^^^^^^^^^^^^^^^^^^^^^^^^^^
+--------------------------
+
+Given a split and a list of columns, the record set provider is responsible for
+delivering data to the Trino execution engine. It creates a ``RecordSet``, which
```

Comment:
In general, avoid using "it" to remove ambiguity about the subject. In this case, I'd suggest "The provider creates..."